servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xPublishController.java

```diff
@@ -263,8 +263,13 @@ private IPath getTempBundlePath(IModule[] module) {
     private String getModuleVersion(IModule module) throws CoreException {
         IProgressMonitor monitor = new NullProgressMonitor();
         IProject project = module.getProject();
-        IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().create(project.getFile(IMavenConstants.POM_FILE_NAME), true, monitor);
-        MavenProject mavenProject = facade.getMavenProject(monitor);
-        return mavenProject.getVersion();
+        if (project == null) {
+        	Activator.getLogger().warning("Cannot determine project associated with module: " + module.getName());
+        	return "";
```

Comment:
I think we should not add another possibility for the return value here. In the calling method we catch CoreException and assign a module version in that case already. I would suggest to throw a CoreException instead of returning an empty string.

Reply:
I was wondering if I should throw an exception or not but I'm not sure that it would cause trouble if the version is not found.
I think that we will need to check that to know if it is better to throw an exception or returning an empty value.

Reply:
Thanks for pointing the code around the method, I decided to group all the information to retrieve module version in the same method, it avoids to have exceptions thrown and catch through different methods.